File: ea-domain/src/main/kotlin/com/nbe3/domain/posts/exception/CommentNotFoundException.java
package com.nbe3.domain.posts.exception;

import com.nbe3.common.exception.DomainException;

public class CommentNotFoundException extends DomainException {
    public static final DomainException EXCEPTION = new CommentNotFoundException();

    private CommentNotFoundException() {
        super(CommentErrorCode.COMMENT_NOT_FOUND);
    }
}

File: ea-domain/src/main/java/com/nbe2/domain/posts/exception/CommentNotFoundException.kt
package com.nbe2.domain.posts.exception

import com.nbe2.common.exception.DomainException

class CommentNotFoundException
private constructor(errorCode: CommentErrorCode) : DomainException(errorCode) {
    companion object {
        val EXCEPTION: DomainException =
                com.nbe2.domain.posts.exception.CommentNotFoundException(
                        CommentErrorCode.COMMENT_NOT_FOUND
                )
    }
}
// object CommentNotFoundException : DomainException(CommentErrorCode.COMMENT_NOT_FOUND)
